toolkit/tools/pkg/imagecustomizerlib/partitionutils.go

```diff
@@ -347,24 +389,207 @@ func findPartition(mountIdType imagecustomizerapi.MountIdentifierType, mountId s
 	return partition, partitionIndex, nil
 }
 
+func convertDevToPartUuid(partitions []diskutils.PartitionInfo, buildDir string) (string, error) {
+	for _, partition := range partitions {
```

Comment:
You do not need this for-loop. Just pass the results of `extractVerityRootPartitionId` up to `findExtendedPartition` and let `findExtendedPartition`'s for-loop find the partition.

Finding the same partition twice in a row is silly.

Reply:
Looks good.